From 64806976b7e312bad3cd4cf67a69b96454210f58 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Sun, 5 Apr 2020 17:36:34 +0300
Subject: [PATCH 32/49] SPL: Adjust voltage for correct DDR initialisation

The voltage values, that PMIC BD71847 produces, differ from required and need to
be adjusted before a memory training procedure.

Rational: i.MX8M Mini Reference Board schematics SCH-31399/SPF-31399 rev C2, p.12.
via: https://www.nxp.com/webapp/Download?colCode=8MMINILPD4-EVK-DESIGNFILES

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 board/compulab/ucm-imx8m-mini/spl.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/board/compulab/ucm-imx8m-mini/spl.c b/board/compulab/ucm-imx8m-mini/spl.c
index c26d3f154e..537eaa465f 100644
--- a/board/compulab/ucm-imx8m-mini/spl.c
+++ b/board/compulab/ucm-imx8m-mini/spl.c
@@ -248,8 +248,14 @@ int power_init_board(void)
 	/* unlock the PMIC regs */
 	pmic_reg_write(p, BD71837_REGLOCK, 0x1);
 
-	/* increase VDD_DRAM to 0.9v for 3Ghz DDR */
-	pmic_reg_write(p, BD71837_BUCK5_VOLT, 0x2);
+	/* decrease VDD_ARM to 0.85V for 1.2GHz operation */
+	/*pmic_reg_write(p, BD71837_BUCK2_VOLT_RUN, 0x0f);*/
+
+	/* increase VDD_SOC to 0.85V for 3Ghz DDR */
+	pmic_reg_write(p, BD71837_BUCK1_VOLT_RUN, 0x0f);
+
+	/* increase VDD_DRAM to 0.975V (9v5 required but not supported)*/
+	pmic_reg_write(p, BD71837_BUCK5_VOLT, 0x83);
 
 	/* lock the PMIC regs */
 	pmic_reg_write(p, BD71837_REGLOCK, 0x11);
-- 
2.11.0

